integration/deploy_manifest_test.go

```diff
@@ -0,0 +1,322 @@
+			t.Fatalf(err.Error())
+		}
+
+		if err := expectAppToBeRunning(releaseName, testNamespace); err != nil {
+			t.Fatalf(err.Error())
+		}
+
+	})
+
+	t.Run("okteto deploy should not build images if already at registry", func(t *testing.T) {
+
+		imageWithDigest, err := registry.GetImageTagWithDigest(expectedImage)
+		if err != nil {
+			t.Fatalf("image is not at registry: %s", err.Error())
+		}
+		sha := strings.SplitN(imageWithDigest, "@", 2)[1]
+
+		output, err := runOktetoDeploy(oktetoPath, repoDir)
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		if err := expectImageFoundSkippingBuild(output); err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		if err := expectHelmUpgrade(output, releaseName, testNamespace, "2"); err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		if err := expectEnvSetting(output, testNamespace, repoDir, sha); err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		if err := expectAppToBeRunning(releaseName, testNamespace); err != nil {
+			t.Fatalf(err.Error())
+		}
+
+	})
+
+	t.Run("okteto deploy --build should force the build even if exist", func(t *testing.T) {
+
+		imageWithDigest, err := registry.GetImageTagWithDigest(expectedImage)
+		if err != nil {
+			t.Fatalf("image is not at registry: %s", err.Error())
+		}
+		sha := strings.SplitN(imageWithDigest, "@", 2)[1]
+
+		output, err := runOktetoDeployForceBuild(oktetoPath, repoDir)
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		if err := expectForceBuild(output); err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		if err := expectHelmUpgrade(output, releaseName, testNamespace, "3"); err != nil {
+			t.Fatalf("expected helm upgrade")
+		}
+
+		if err := expectEnvSetting(output, testNamespace, repoDir, sha); err != nil {
+			t.Fatalf("expected to environment variables to be set")
+		}
+
+		if err := expectAppToBeRunning(releaseName, testNamespace); err != nil {
+			t.Fatalf("expected the app to be running with exact content")
+		}
+
+	})
+}
+
+func runOktetoDeploy(oktetoPath, repoDir string) (string, error) {
+	cmd := exec.Command(oktetoPath, "deploy", "-l", "debug")
+	cmd.Env = append(os.Environ(), "OKTETO_ENABLE_MANIFEST_V2=true")
+	cmd.Dir = repoDir
+	o, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("okteto deploy failed: \nerror: %s \noutput: %s", err.Error(), string(o))
+	}
+	return string(o), nil
+}
+
+func runOktetoDeployForceBuild(oktetoPath, repoDir string) (string, error) {
+	cmd := exec.Command(oktetoPath, "deploy", "--build", "-l", "debug")
+	cmd.Env = append(os.Environ(), "OKTETO_ENABLE_MANIFEST_V2=true")
+	cmd.Dir = repoDir
+	o, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("okteto deploy --build failed: \nerror: %s \noutput: %s", err.Error(), string(o))
+	}
+	return string(o), nil
+}
+
+func expectBuiltImageNotFound(output string) error {
+	if ok := strings.Contains(output, "image not found, building image"); !ok {
+		return fmt.Errorf("expected image not found, building image")
```

Comment:
If you don't need string formatting just call `errors.New` It's odd that it is this way but it is.